### PR TITLE
[feat-5532]: write Attachments with permission

### PR DESCRIPTION
**Pull request by SireeKoolenWijkstra**
**only users with right permissions are able to see/edit/delete attachments**

- Only users with permission sia_add_attachment will see the button Bestand toevoegen
- Only users with permission sia_change_attachment will see the delete button and the edit button.
- Permissions sia_change_attachment and sia_add_attachment are added to permissions for Attachments
- Permissions are moved to constants file in IncidentDetail directory.

Ticket: [SIG-5532](https://gemeente-amsterdam.atlassian.net/browse/SIG-5532)

## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

[SIG-5532]: https://gemeente-amsterdam.atlassian.net/browse/SIG-5532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by OscarBakker** [comment]
would call it not crud but write Attachments with permission as title

File: src/signals/incident-management/containers/IncidentDetail/components/Attachments/Attachments.test.tsx [hunk 1]
```diff
@@ -763,6 +788,54 @@ describe('Attachments', () => {
...
+      // expect(
+      //   screen.queryByRole('checkbox', { name: 'Openbaar tonen' })
+      // ).toBeInTheDocument()
+      //
+      // expect(screen.queryByText(/Onderschrift/)).not.toBeInTheDocument()
+      //
+      // userEvent.click(screen.getByRole('checkbox', { name: 'Openbaar tonen' }))
+      //
+      // expect(screen.getByText(/Onderschrift/)).toBeInTheDocument()
+    })
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/IncidentDetail/components/Attachments/Attachments.test.tsx:838]: Commented code...

**Review by vdegraaf — COMMENTED** [review]
If I test correctly, the attachments are not shown anymore when the user does not have writing permissions. I don't think that is intended.  -> yeah, apparently is SIA_WRITE shared with everybody. The rights have changed (see Davids comments below). I've got some difficulties testing, so I hope it's ok now.

File: src/signals/incident-management/containers/IncidentDetail/constants.ts [hunk 2]
```diff
@@ -38,3 +38,9 @@ export const PREVIEW =
 export const EDIT = 'sia/incidentManagement/containers/IncidentDetail/EDIT'
 export const EXTERNAL =
   'sia/incidentManagement/containers/IncidentDetail/EXTERNAL'
+
+export const DELETE_CHILD = 'sia_delete_attachment_of_child_signal'
+export const DELETE_NORMAL = 'sia_delete_attachment_of_normal_signal'
+export const DELETE_OTHER = 'sia_delete_attachment_of_other_user'
+export const DELETE_PARENT = 'sia_delete_attachment_of_parent_signal'
+export const SIA_WRITE = 'sia_write'
```
Comment (vanbuiten) [inline src/signals/incident-management/containers/IncidentDetail/constants.ts:46]: This is not the correct permission. It also should not be 1 but 2 permissions that must be checked.

**Review by vanbuiten — CHANGES_REQUESTED** [review]
See Jira (SIG-5533 and SIG-5509) for the naming of the new permissions. Please implement these new permissions instead of `sia_write`. There are 2 separate permissions for adding and updating Attachments.

File: src/signals/incident-management/containers/IncidentDetail/components/Attachments/Attachments.test.tsx (hunk 1 above)
Comment (SireeKoolenWijkstra) [reply src/signals/incident-management/containers/IncidentDetail/components/Attachments/Attachments.test.tsx]: Removed it.

File: src/signals/incident-management/containers/IncidentDetail/constants.ts (hunk 2 above)
Comment (SireeKoolenWijkstra) [reply src/signals/incident-management/containers/IncidentDetail/constants.ts]: I've incorporated the new permissions.

**Review by vdegraaf — COMMENTED** [review]
The attachments still dissapear when I don't have the CHANGE permission. See previous comment. Is that intended? Don't think so since I'm able to upload an attachment.

File: src/signals/incident-management/containers/IncidentDetail/constants.ts [hunk 3]
```diff
@@ -38,3 +38,10 @@ export const PREVIEW =
 export const EDIT = 'sia/incidentManagement/containers/IncidentDetail/EDIT'
 export const EXTERNAL =
   'sia/incidentManagement/containers/IncidentDetail/EXTERNAL'
+
+export const DELETE_CHILD = 'sia_delete_attachment_of_child_signal'
+export const DELETE_NORMAL = 'sia_delete_attachment_of_normal_signal'
+export const DELETE_OTHER = 'sia_delete_attachment_of_other_user'
+export const DELETE_PARENT = 'sia_delete_attachment_of_parent_signal'
+export const SIA_CHANGE_ATTACHMENT = 'sia_change_attachment'
+export const SIA_ADD_ATTACHMENT = 'sia_add_attachment'
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/signals/incident-management/containers/IncidentDetail/constants.ts]: Did it

**Review by vdegraaf — APPROVED** [review]
LGTM
Just the ticket is not up to date.

**Pull request merged**